src/build/arg/mod.rs

```diff
@@ -2581,38 +2581,60 @@ impl<'help> Arg<'help> {
     ///         .long("opt"))
     ///     .arg(Arg::new("other")
     ///         .long("other")
-    ///         .default_value_if("opt", Some("special"), "default"))
+    ///         .default_value_if("opt", Some("special"), Some("default")))
     ///     .get_matches_from(vec![
     ///         "prog", "--opt", "hahaha"
     ///     ]);
     ///
     /// assert_eq!(m.value_of("other"), None);
     /// ```
+    ///
+    /// We can also remove a default if the flag was set by setting `default` to `None`.
```

Comment:
This needs better phrasing. It does not explain things properly as it is right now.